### PR TITLE
Migrate/rpn1 61 code from engine

### PR DESCRIPTION
**Pull request by Sajjon**
### Change
Readding constructor to create a `RadixAddress` from a `Universe` and `ECPublicKey`, without adding `Universe` to this repo, by introducing a trivial interface called `Magical`.

Prefixing all unit tests with `from_engine___`, which is a **temporary** prefix, to be removed after having migrated tests from Java library.

### Engine:
Relevant corresponding branch in engine:
https://github.com/radixdlt/radix-engine-library/tree/improvement/rpn1-61-migrate-code-to-java-common

### Status
All tests are passing in this repo and in engine, concluding extraction of code from Engine to common.

#### Next Step
Next step is to migrate equivalent code from Java library and all unit tests (prefixing them with `"from_library___"`)

### CODE REVIEW

File: src/main/java/com/radixdlt/identifiers/Magical.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.identifiers;
+
+public interface Magical {
```
Comment (clstrfsck) [inline src/main/java/com/radixdlt/identifiers/Magical.java]: This seems a bit heavy handed to me.  For now I think the simplest thing that would work would be to pass in the magic number to constructors or factory methods.  Not really convinced that we need an interface for something that returns a number.

File: src/main/java/com/radixdlt/identifiers/RadixAddress.java [hunk 2]
```diff
@@ -30,7 +30,7 @@
 /**
  * base58 address based on a public key
  */
-public final class RadixAddress {
+public final class RadixAddress implements Magical {
```
Comment (clstrfsck) [inline src/main/java/com/radixdlt/identifiers/RadixAddress.java]: If we are going to go with the interface, then getMagic() needs to have an @Override annotation, which your IDE should have warned you about.

**Comment by clstrfsck** [comment]
Also could have a look at setting up travis for CI as well.  See .travis.yml file in radix-engine-library (for example).

File: src/main/java/com/radixdlt/identifiers/Magical.java (hunk 1 above)
Comment (Sajjon) [reply src/main/java/com/radixdlt/identifiers/Magical.java]: Yeah sure! Removed :)

**Pull request merged**